services/blockstorage/test/block_sync_test.go

```diff
@@ -0,0 +1,441 @@
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func generateBlockAvailabilityRequestInput(lastCommittedBlockHeight primitives.BlockHeight, senderPublicKey primitives.Ed25519PublicKey) *gossiptopics.BlockAvailabilityRequestInput {
+	return &gossiptopics.BlockAvailabilityRequestInput{
+		Message: &gossipmessages.BlockAvailabilityRequestMessage{
+			SignedBatchRange: (&gossipmessages.BlockSyncRangeBuilder{
+				BlockType:                gossipmessages.BLOCK_TYPE_BLOCK_PAIR,
+				LastCommittedBlockHeight: lastCommittedBlockHeight,
+			}).Build(),
+			Sender: (&gossipmessages.SenderSignatureBuilder{
+				SenderPublicKey: senderPublicKey,
+			}).Build(),
+		},
+	}
+}
+
+func TestSyncSourceHandlesBlockAvailabilityRequest(t *testing.T) {
+	// FIXME user WithContext everywhere
+	//test.WithContext(func(ctx context.Context) {
+
+	driver := NewDriver()
+
+	driver.expectCommitStateDiffTimes(2)
+
+	driver.commitBlock(builders.BlockPair().WithHeight(primitives.BlockHeight(1)).WithBlockCreated(time.Now()).Build())
+	driver.commitBlock(builders.BlockPair().WithHeight(primitives.BlockHeight(2)).WithBlockCreated(time.Now()).Build())
+
+	senderKeyPair := keys.Ed25519KeyPairForTests(9)
+
+	input := generateBlockAvailabilityRequestInput(primitives.BlockHeight(0), senderKeyPair.PublicKey())
+	response := &gossiptopics.BlockAvailabilityResponseInput{
+		RecipientPublicKey: senderKeyPair.PublicKey(),
+		Message: &gossipmessages.BlockAvailabilityResponseMessage{
+			SignedBatchRange: (&gossipmessages.BlockSyncRangeBuilder{
+				BlockType:                gossipmessages.BLOCK_TYPE_BLOCK_PAIR,
+				LastBlockHeight:          primitives.BlockHeight(2),
+				FirstBlockHeight:         primitives.BlockHeight(1),
+				LastCommittedBlockHeight: primitives.BlockHeight(2),
+			}).Build(),
+			Sender: (&gossipmessages.SenderSignatureBuilder{
+				SenderPublicKey: driver.config.NodePublicKey(),
+			}).Build(),
+		},
+	}
+
+	driver.gossip.When("SendBlockAvailabilityResponse", response).Return(nil, nil).Times(1)
+
+	_, err := driver.blockStorage.HandleBlockAvailabilityRequest(input)
+	require.NoError(t, err)
+
+	// FIXME remove sleep
+	time.Sleep(1 * time.Millisecond)
+
+	driver.verifyMocks(t)
+}
+
+func TestSyncSourceIgnoresBlockAvailabilityRequestIfNoBlocksWereCommitted(t *testing.T) {
+	driver := NewDriver()
+
+	senderKeyPair := keys.Ed25519KeyPairForTests(9)
+	input := generateBlockAvailabilityRequestInput(primitives.BlockHeight(2), senderKeyPair.PublicKey())
+
+	driver.gossip.When("SendBlockAvailabilityResponse", mock.Any).Return(nil, nil).Times(0)
+
+	_, err := driver.blockStorage.HandleBlockAvailabilityRequest(input)
+	require.NoError(t, err)
+
+	// FIXME remove sleep
+	time.Sleep(1 * time.Millisecond)
+
+	driver.verifyMocks(t)
+}
+
+func TestSyncSourceIgnoresBlockAvailabilityRequestIfPetitionerIsFurtherAhead(t *testing.T) {
```

Comment:
We have a very large task of moving many of these tests to unit tests

Let's focus on this task for your next PR

I don't want the next PR to add functionality or features, just convert to unit tests